RadixWallet/Clients/OnLedgerEntitiesClient/Helpers/OnLedgerEntitiesClient+CreateEntity.swift

```diff
@@ -150,7 +150,7 @@ extension OnLedgerEntitiesClient {
 		guard let xrdResource = item
 			.fungibleResources?
 			.items
-			.first(where: { $0.resourceAddress == xrdAddress })
+			.first(where: { $0.resourceAddress == xrdAddress.address })
```

Comment:
Gateway returns string... will be VERY nice when Gateway Client lives in Sargon, and we have a wrapper around it translating all Gateway raw models into "Ornamented" models, where all addresses are translated into our own types, and not work with raw Strings...

Reply:
Yes my plan is one layer which turns String into Addresses and Decimal192'etc real Sargon types - without performing any application logic mapping - and upon making network call, the response is mapped to those Sargonified types. 

This has the advantage that we catch any bugs in GW early, regarding the fetch as failure if we failed to deserialise and try into Sargonified. 

What we later do with GW types - is application logic. We might chose to save a subset of the data in some
Local cache - similar to what is call OnLedgerEntity in iOS wallet. That code will be trivial, since it is about picking property from already parsed and Sargonified types.

Reply:
But I just realised that we can do that we a script that modifies the generated models. 

The openapi generator is completely broken for Rust anyway, so we need to modify the schema file, either we modified the schema file further to generate Decimal192 and typed addresses instead of Strings- not trivially done, but the modifications we need to do just to get models to work is hard however we do it. Or we make a simple post processor script which upgrade strings to Decimal192 and typed addresses.

Reply:
yes, absolutely, if we are to do this it needs to be automated. Otherwise we might end up maintaining by hand lots of models which are not even used by Sargon/Wallet.